### PR TITLE
Add donation amount form on the project page

### PR DESCRIPTION
**Pull request by cmc333333**
There's a radio selector with four preset options (10, 25, 50, all) and a custom option. Not particularly styled.

### CODE REVIEW

File: peacecorps/peacecorps/forms.py [hunk 1]
```diff
@@ -133,3 +134,41 @@ def clean(self):
...
+            if self.cleaned_data.get('presets') == 'preset-' + str(amt):
+                return Decimal(amt)
+        if self.cleaned_data.get('presets') == 'preset-all':
+            if not self.fund:
+                raise ValidationError('Missing fund')
+            else:
+                remaining_amount = self.fund.fundgoal - self.fund.fundcurrent
+                if remaining_amount < 100:  # cents
+                    raise ValidationError('Must be >= 1.00')
+                elif remaining_amount > 1000000:
```
Comment (khandelwal) [inline peacecorps/peacecorps/forms.py]: Let's add a comment to explain why we're doing this.

File: peacecorps/peacecorps/tests/test_views.py [hunk 2]
```diff
@@ -175,3 +174,30 @@ def test_issue_rendering(self):
     def test_project_rendering(self):
         response = self.client.get('/donate/project/test-project')
         self.assertEqual(response.status_code, 200)
+
```
Comment (khandelwal) [inline peacecorps/peacecorps/tests/test_views.py:177]: Thank you for tests!

File: peacecorps/peacecorps/forms.py [hunk 3]
```diff
@@ -133,3 +134,43 @@ def clean(self):
...
+        if self.cleaned_data.get('presets') == 'preset-all':
+            if not self.fund:
+                raise ValidationError('Missing fund')
+            else:
+                remaining_amount = self.fund.fundgoal - self.fund.fundcurrent
+                # We're circumventing the normal bounds checks, so we must
+                # enforce them here
+                if remaining_amount < 100:  # cents
+                    raise ValidationError('Must be >= 1.00')
+                elif remaining_amount > 1000000:
```
Comment (khandelwal) [inline peacecorps/peacecorps/forms.py]: @seanherron says this should be >= 1000000

**Comment by seanherron** [comment]
Question - can we show the actual amount you would be paying next to the "fund the remaining amount" option?

**Comment by khandelwal** [comment]
So, in this we're making putting >= $10,000 as a validation error for the form. That's fine.  However, should we be handling that case more gracefully? Are there specific instructions we should direct the user towards for large donations?

Asking...for a friend.

**Comment by cmc333333** [comment]
@seanherron @khandelwal - both great questions; bring them up with design.

I don't believe either concern should prevent this pull request, a first draft, from being merged.

**Pull request merged**